Make Frame::get<T> throw on non-existing collections instead of returning an empty collection (#868)
AIDASoft/podio: doc/examples.md

Collections can be retrieved explicitly:

```cpp
    try {
      auto& hits = frame.get<HitCollection>("hits");
    } catch (const std::runtime_error&) {

>       // The collection "hits" has not been available as type HitCollection

Reviewer: ```suggestion
      // The collection "hits" does not exist or isn't a HitCollection
```
